frb_example/pure_dart/rust/src/api.rs

```diff
@@ -1005,43 +1029,82 @@ pub fn run_enum_opaque(opaque: EnumOpaque) -> String {
     }
 }
 
-pub fn run_opaque(opaque: Opaque<HideData>) -> String {
+pub fn run_opaque(opaque: RustOpaque<HideData>) -> String {
     opaque.hide_data()
 }
 
-pub fn run_opaque_with_delay(opaque: Opaque<HideData>) -> String {
+pub fn run_opaque_with_delay(opaque: RustOpaque<HideData>) -> String {
     sleep(Duration::from_millis(1000));
     opaque.hide_data()
 }
 
-pub fn opaque_array() -> [Opaque<HideData>; 2] {
-    [Opaque::new(HideData::new()), Opaque::new(HideData::new())]
+pub fn opaque_array() -> [RustOpaque<HideData>; 2] {
+    [
+        RustOpaque::new(HideData::new()),
+        RustOpaque::new(HideData::new()),
+    ]
 }
 
-pub fn opaque_array_run(data: [Opaque<HideData>; 2]) {
+pub fn opaque_array_run(data: [RustOpaque<HideData>; 2]) {
     for i in data {
         i.hide_data();
     }
 }
 
-pub fn opaque_vec() -> Vec<Opaque<HideData>> {
-    vec![Opaque::new(HideData::new()), Opaque::new(HideData::new())]
+pub fn opaque_vec() -> Vec<RustOpaque<HideData>> {
+    vec![
+        RustOpaque::new(HideData::new()),
+        RustOpaque::new(HideData::new()),
+    ]
 }
 
-pub fn opaque_vec_run(data: Vec<Opaque<HideData>>) {
+pub fn opaque_vec_run(data: Vec<RustOpaque<HideData>>) {
     for i in data {
         i.hide_data();
     }
 }
 
 pub fn create_nested_opaque() -> OpaqueNested {
     OpaqueNested {
-        first: Opaque::new(HideData::new()),
-        second: Opaque::new(HideData::new()),
+        first: RustOpaque::new(HideData::new()),
+        second: RustOpaque::new(HideData::new()),
     }
 }
 
 pub fn run_nested_opaque(opaque: OpaqueNested) {
     opaque.first.hide_data();
     opaque.second.hide_data();
 }
+
+pub fn unwrap_rust_opaque(opaque: RustOpaque<HideData>) -> String {
+    let hide_data = opaque.try_unwrap().unwrap();
+    hide_data.hide_data()
+}
+
+pub fn unwrap_dart_opaque(opaque: DartOpaque) -> SyncReturn<String> {
+    #[cfg(target_family = "wasm")]
+    {
+        let js = opaque.try_unwrap().unwrap();
+        SyncReturn(js.as_string().unwrap())
+    }
+    #[cfg(not(target_family = "wasm"))]
+    {
+        let handle = opaque.try_unwrap().unwrap();
+        unsafe { Dart_DeletePersistentHandle_DL_Trampolined(handle) };
+        SyncReturn("Test".to_owned())
+    }
+}
+
+pub fn panic_unwrap_dart_opaque(opaque: DartOpaque) -> String {
+    #[cfg(target_family = "wasm")]
+    {
+        let js = opaque.try_unwrap().unwrap();
+        js.as_string().unwrap()
+    }
+    #[cfg(not(target_family = "wasm"))]
+    {
+        let handle = opaque.try_unwrap().unwrap();
+        unsafe { Dart_DeletePersistentHandle_DL_Trampolined(handle) };
```

Comment:
imho, unwrap_dart_opaque vs panic_unwrap_dart_opaque is only different in whether it is a SyncReturn? thus not sure why "panic"

Reply:
exactly in SyncReturn. Since in this case it is the same thread that the DartObject (JsValue) was created on, we can do an unwrap

Reply:
I see. Then it looks a bit confusing for readers